lib/main/atom/commands/refactorCode.tsx

```diff
@@ -0,0 +1,131 @@
+import {addCommand, Dependencies} from "./registry"
+import {commandForTypeScript, getFilePathPosition, spanToRange} from "../utils"
+import {selectListView} from "../views/simpleSelectionView"
+import * as etch from "etch"
+import {HighlightComponent} from "../views/highlightComponent"
+import * as protocol from "typescript/lib/protocol"
+import * as atom from "atom"
+import {TypescriptServiceClient} from "../../../client/client"
+
+interface RefactorAction {
+  refactorName: string
+  refactorDescription: string
+  actionName: string
+  actionDescription: string
+  inlineable: boolean
+}
+
+addCommand("atom-text-editor", "typescript:refactor-selection", deps => ({
+  description: "Get a list of applicable refactors to selected code",
+  async didDispatch(e) {
+    if (!commandForTypeScript(e)) {
+      return
+    }
+
+    const editor = e.currentTarget.getModel()
+    const location = getFilePathPosition(editor)
+    if (!location) {
+      e.abortKeyBinding()
+      return
+    }
+    const selection = editor.getSelectedBufferRange()
+    if (selection.isEmpty()) {
+      e.abortKeyBinding()
+      return
+    }
+    const client = await deps.getClient(location.file)
+
+    const fileRange: protocol.FileRangeRequestArgs = {
+      file: location.file,
+      startLine: selection.start.row + 1,
+      startOffset: selection.start.column + 1,
+      endLine: selection.end.row + 1,
+      endOffset: selection.end.column + 1,
+    }
+
+    const actions = await getApplicableRefactorsActions(client, fileRange)
+
+    if (actions.length === 0) {
+      // TODO Show a "no applicable refactors here" message
+      e.abortKeyBinding()
```

Comment:
Really shouldn't `abortKeyBinding` here if we're showing the message. The idea of `abortKeyBinding` is to behave as if this handler wasn't executed in the first place (so that other handlers might do their work, e.g. if different commands are bound to the same key). If you're showing the message, that's a side-effect that shows that the handler was executed, so it really makes little sense pretending it wasn't.

Reply:
Oh, sure. That block should be replaced with that of show message.